loopy/kernel/array.py

```diff
@@ -659,7 +659,7 @@ class ArrayBase(ImmutableRecord):
 
     def __init__(self, name, dtype=None, shape=None, dim_tags=None, offset=0,
             dim_names=None, strides=None, order=None, for_atomic=False,
-            target=None, alignment=None,
+            target=None, alignment=None, tags=None,
```

Comment:
`ArrayBase` should inherit from `pytools.tag.Taggable`, which needs to be created.